tests/integration/test_engine_integration.py

```diff
@@ -322,3 +308,435 @@ def test_measurefock_shots_exception(self, setup_eng):
                            match=r"""(Measure|MeasureFock) has not been implemented in {} """
                                   """for the arguments {{'shots': {}}}""".format(backend_name, shots)):
             eng.run(p1, shots=shots).samples
+
+class TestResults:
```

Comment:
Note: I have a feeling some of these tests could be simplified/unified, perhaps with a `pytest.mark.parameterize`? Suggestions welcome